### ISSUE TITLE
Allow custom alignment for `AlignedVec`

### ISSUE THREAD

**Issue by overlookmotel** [rkyv/rkyv#354]
`AlignedVec` currently has a set alignment of 16.

In some cases, you might wish to have an `AlignedVec` aligned on higher alignment. e.g. to be able to store objects with a higher alignment requirement than 16.

That'd be the most common use case, though my particular use case is actually a bit different. I'm using RKYV's archive format as a serialization format for interop between Rust and Javascript with a serializer/deserializer for the format written in JS (https://github.com/swc-project/swc/issues/2175).

To optimize the code on JS side, it'd be useful to be able to give the buffer a very high alignment (like 65536), to simplify the maths when calculating pointers.

I'm imagining this being supported through a const parameter:

```rust
pub struct AlignedVecCustom<const ALIGNMENT: usize> { /* ... */ }

impl<const ALIGNMENT: usize> AlignedVecCustom<ALIGNMENT> {
  const ALIGNMENT: usize = ALIGNMENT;
  /* ... */
}
```

Current `AlignedVec` would be defined as:

```rust
pub type AlignedVec = AlignedVecCustom<16>;
```

I've put together the beginning of a mock-up as PR #353.

What do you think? I would totally understand if you think this is too niche a use case to support.

**Comment by djkoloski** [comment]
I think this would be fine to do. Adding a type alias for compatibility should keep semver intact.

**Comment by Alxandr** [comment]
Feel free to just take the code from my aligned buffer crates.

**Comment by djkoloski** [comment]
Closed by #353
